classes/context/PKPSectionDAO.inc.php

```diff
@@ -83,9 +83,39 @@ function deleteByContextId($contextId) {
 
 	/**
 	 * Retrieve all sections for a context.
+	 * @param $contextId int
+	 * @param $rangeInfo DBResultRange optional
+	 * @param $submittableOnly boolean ooptional. Whether to return only sections
+	 *  that can be submitted to by anyone.
 	 * @return DAOResultFactory containing Sections ordered by sequence
 	 */
-	abstract function getByContextId($contextId, $rangeInfo = null);
+	public function getByContextId($contextId, $rangeInfo = null, $submittableOnly = false) {
```

Comment:
This is currently abstract in `pkp-lib` because `ojs` (`SectionDAO`) and `omp` (`SeriesDAO`) offer different implementations of it; you'd need to consider the implications for OMP.

Reply:
I suggest we leave the method abstract but add the `$submittableOnly` parameter. The implementation can then be extended in OMP to consider the editor restricted flag as it is done in OJS. If you agree I can provide a PR for OMP.

Reply:
Thanks, that sounds sensible to me.

Reply:
Done, method remains abstract and `$submittableOnly` parameter added.